tests/exponential_backoff_retry_test.c

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include <aws/io/retry_strategy.h>
+
+#include <aws/testing/aws_test_harness.h>
+
+#include <aws/common/clock.h>
+#include <aws/common/condition_variable.h>
+
+#include <aws/io/event_loop.h>
+
+struct exponential_backoff_test_data {
+    size_t retry_count;
+    size_t client_error_count;
+    int failure_error_code;
+    struct aws_mutex mutex;
+    struct aws_condition_variable cvar;
+};
+
+static void s_too_many_retries_test_on_retry_ready(struct aws_retry_token *token, int error_code, void *user_data) {
```

Comment:
Can you just add a comment on each test case to explain what you're testing for briefly, just to be a bit more clear?